electron_app/DownloadManager.js

```diff
@@ -2,22 +2,29 @@ const mtd = require('zeltice-mt-downloader');
 const path = require('path');
 const os = require('os');
 
-const handleFolder = require('./helpers/handleDisk');
+const { handleFolder, handleDiskSpace } = require('./helpers/handleDisk');
 
 let downElements = [];
 
 const downloadManager = (items, access_token) => {
-  let filePath = '';
-  const options = setHeader(access_token);
-  items.forEach(item => {
-    if (item.preserveStructure) {
-      const structurePath = item.path.substring(item.path.lastIndexOf('/'), 0);
-      handleFolder(path.join(item.destination, structurePath)).then(result =>
-        processDownload(access_token, item, result)
-      );
-    } else {
-      processDownload(access_token, item, item.destination);
-    }
+  return new Promise((res, rej) => {
+    let filePath = '';
+    const options = setHeader(access_token);
+    handleDiskSpace(items).then(saveConditions => {
+      if (saveConditions.diskSpace && saveConditions.allowed) {
```

Comment:
Since the body is only executed when both these cases are true, and since this is the only use of `handleDiskSpace()`, why not change it to not call the `then()` method at all if either condition is false? That would avoid the need to return boolean results from this method.

Reply:
I understand what you mean. Although there's a reason why the method returns an object with two booleans, which is that we need some type of flag to identify the status of each conditions (free disk space, and write permissions) so we can later display its proper warning in the UI, if you take a look to `main.js` here we send this boolean object to Angular's part. More specifically to `file-modal.ts`.

Reply:
OK. Since `rej` was unused I didn't see how `saveConditions` would be returned in the negative case. Another approach would be to return an error string since either error condition is a failure and it doesn't matter to the caller which failure it is.

Reply:
i have changed the return object to a more generic one, instead of  `[hasEnoughSpace :Boolean, allowed: Boolean}` this is now `{hasError: Boolean, errorMessage: String}`. That way, there would be only one flag for an error and a better handling of the specific error. 
Regarding `rej()` function for the promise, i'm adding a try/catch block to get any error coming from there, and login it as a console.error.